### ISSUE TITLE
RFE: sup-sync should give the filename on errors, not just the message id

### ISSUE THREAD

**Issue by fxkr** [sup-heliotrope/sup#480]
That'd save me from waiting 4 minutes for grep to finish searching through my 170000 mails...

```
## scanned 6848m (~4%) @ 75.5m/s. 0:01:30 elapsed, ~0:36:03 remaining
[2015-09-26 13:12:00 +0200] WARNING: problem reading message 012.345@xyz
/home/fxkr/.gem/ruby/gems/gpgme-2.0.10/lib/gpgme/ctx.rb:333:in `get_key': EOFError (EOFError)
```

(I'll open a seperate issue for the actual error.)

**Comment by rakoo** [comment]
The warning is correctly displayed but the debug could be interesting. Could you try applying these changes ?

```patch
diff --git a/lib/sup/message.rb b/lib/sup/message.rb
index 59ef7cd..509b7de 100644
--- a/lib/sup/message.rb
+++ b/lib/sup/message.rb
@@ -265,14 +265,14 @@ class Message
         message_to_chunks rmsg
       rescue SourceError, SocketError, RMail::EncodingUnsupportedError => e
         warn "problem reading message #{id}"
-        debug "could not load message: #{location.inspect}, exception: #{e.inspect}"
+        warn "could not load message: #{location.inspect}, exception: #{e.inspect}"
 
         [Chunk::Text.new(error_message.split("\n"))]
 
       rescue Exception => e
 
         warn "problem reading message #{id}"
-        debug "could not load message: #{location.inspect}, exception: #{e.inspect}"
+        warn "could not load message: #{location.inspect}, exception: #{e.inspect}"
 
         raise e
 ```

**Comment by fxkr** [comment]
* Ok, patch applied. Output (with some sensitive info nulled):

```
[2015-09-26 14:54:33 +0200] WARNING: problem reading message 00000000.0000000@example.com
[2015-09-26 14:54:33 +0200] WARNING: could not load message: #<Redwood::Location:0x000000089d1180 @source=#<Redwood::Maildir:0x000000019370f0 @uri="maildir:///~/.mail", @usual=true, @archived=false, @id=1, @poll_lock=#<Monitor:0x00000001935868 @mon_owner=#<Thread:0x00000000a7f328 run>, @mon_count=1, @mon_mutex=#<Mutex:0x000000019356d8>>, @expanded_uri="maildir:////home/fxkr/.mail", @path="//home/fxkr/.mail", @sync_back=false, @dir="//home/fxkr/.mail", @labels=#<Set: {}>, @mutex=#<Mutex:0x00000001934530>, @ctimes={"cur"=>2015-09-26 12:44:24 +0200, "new"=>2015-09-26 14:46:25 +0200}>, @info="new/0000000000.M000000P00000Q00000000000000000000.localhost">, exception: #<EOFError: EOFError>
/home/fxkr/.gem/ruby/gems/gpgme-2.0.10/lib/gpgme/ctx.rb:333:in `get_key': EOFError (EOFError)
	from /home/fxkr/.gem/ruby/gems/gpgme-2.0.10/lib/gpgme/signature.rb:43:in `block in from'
...
```

* "Bad" messages seem to never come alone (it seems like it's usually sender-specific, so if its someone I have a lot of mail from...), so it'd be nice if sup-sync had an option to log the message IDs and paths to a file and carry on.

* Also, here's are some warning message where I got neither a message ID nor a path, where I'd really like to know the message ID:

```
[2015-09-26 14:48:14 +0200] WARNING: multipart/encrypted with 1 parts (expecting 2)
...
[2015-09-26 14:52:31 +0200] WARNING: content_type image/svg+xml is not safe, changed to application/octet-stream
```

**Comment by rakoo** [comment]
I pushed #481 that should display enough information for the user to know where the problem occurs. The problem we have here is that although I believe everyone uses Maildirs, sup only considers sources as generic key => message stores; in the mbox implementation the key is actually an offset in the file.

Regarding logging, you should be able to run something like

```shell
sup-sync 2> log_file
```

to log all the errors.

**Comment by fxkr** [comment]
I opened #483 for crash itself.

**Comment by gauteh** [comment]
This should be fixed in #481.
